pylint/checkers/variables.py

```diff
@@ -624,6 +645,42 @@ def get_next_to_consume(self, node: nodes.Name) -> list[nodes.NodeNG] | None:
 
         return found_nodes
 
+    def _uncertain_nodes_nonlocal(
+        self,
+        found_nodes: list[nodes.NodeNg],
+        node: nodes.Name,
+        visited_nonlocal_nodes: list[list[nodes.Nonlocal]],
+    ) -> list[nodes.NodeNg]:
+        """Return found nodes that can be excluded for checks for cases
+        where `node` is already defined by a nonlocal statement.
+        """
+        uncertain_nodes = []
+        node_frame = node.frame()
+
+        for found_node in found_nodes:
+            # Determine the frame for the current found node
+            found_node_frame = (
+                found_node.parent.frame()
+                if isinstance(found_node, nodes.FunctionDef)
+                else found_node.frame()
+            )
+
+            # If both nodes in same frame, check for nonlocal definition in their frame
+            # Else, check for nonlocal definitions in visited frames
+            if any(
+                node.name in nonlocal_node.names and _is_before(nonlocal_node, node)
+                for scope in visited_nonlocal_nodes
+                if (found_node_frame != node_frame)
+                or (
+                    found_node_frame == node_frame
+                    and (scope and scope[0].frame() == node_frame)
+                )
+                for nonlocal_node in scope
+            ):
+                uncertain_nodes.append(found_node)
+
+        return uncertain_nodes
+
```

Comment:
@jacobtylerwalls i was wondering if it would be better to have this nonlocal check in `VariablesChecker` instead of consumer. if it is in the checker, i suppose we need to check `consumer.consumed_uncertain` on top of `found_nodes` because of consumer filtering